src/itensor.jl

```diff
@@ -1244,10 +1244,22 @@ end
 
 function (A::ITensor + B::ITensor)
   if ndims(A) == 0 && ndims(B) > 0
-    return copy(B)
+    if store(A) isa NDTensors.Empty
+      return copy(B)
+    else # A is a scalar
+      C = copy(B)
+      C .+= A[]
```

Comment:
We decided that we don't want to allow people to do `ITensor(2) + ITensor(i, j)`, so this case should error (that case should get caught by the `ndims(A) != ndims(B)` check below).